test/lib/tasks/seeds_test.rb

```diff
@@ -1,72 +1,108 @@
+      end
     end
 
-    test 'with SEED_LOCATION' do
-      with_env('SEED_LOCATION' => 'test_location_seed') { seed }
-      assert Location.find_by_title('test_location_seed').present?
-    end
+    class PulpProxyTest < ActiveSupport::TestCase
+      test "Make sure Pulp Proxy features exist" do
+        load "#{Katello::Engine.root}/db/seeds.d/104-proxy.rb"
 
-    test 'without SEED_LOCATION' do
-      seed
-      # check that default_location_subscribed_hosts gets set
-      assert_equal Setting.find_by_name('default_location_subscribed_hosts').value, Location.first.title
-      # check that default_location_puppet_content gets set
-      assert_equal Setting.find_by_name('default_location_puppet_content').value, Location.first.title
+        assert Feature.find_by_name('Pulp').present?
+        assert Feature.find_by_name('Pulp Node').present?
+      end
     end
-  end
 
-  class PulpProxyTest < SeedsTest
-    test "Make sure Pulp Proxy features exist" do
-      seed
-      assert Feature.find_by_name('Pulp').present?
-      assert Feature.find_by_name('Pulp Node').present?
+    class MailNotificationsTest < ActiveSupport::TestCase
+      test "Make sure mail notification got setup" do
+        load "#{Katello::Engine.root}/db/seeds.d/106-mail_notifications.rb"
+
+        assert MailNotification[:host_errata_advisory]
+        assert MailNotification[:promote_errata]
+        assert MailNotification[:sync_errata]
+      end
     end
-  end
 
-  class MailNotificationsTest < SeedsTest
-    test "Make sure mail notification got setup" do
-      seed
-      assert MailNotification[:host_errata_advisory]
-      assert MailNotification[:promote_errata]
-      assert MailNotification[:sync_errata]
+    class SubscriptionBookmarkstest < ActiveSupport::TestCase
+      test "Ensure hypervisor bookmark is created" do
+        load "#{Katello::Engine.root}/db/seeds.d/108-subcription-bookmarks.rb"
+
+        refute Bookmark.where(:name => "list hypervisors").empty?
+      end
     end
-  end
 
-  class SubscriptionBookmarkstest < SeedsTest
-    test "Ensure hypervisor bookmark is created" do
-      seed
-      refute Bookmark.where(:name => "list hypervisors").empty?
+    class HttpProxyTest < ActiveSupport::TestCase
+      def setup
+        Setting[:content_default_http_proxy] = ""
+      end
+
+      def run_proxy_seed
+        load "#{Katello::Engine.root}/db/seeds.d/115-http_proxy.rb"
+      end
+
+      test "Clears out default setting if not configured" do
+        SETTINGS[:katello][:cdn_proxy] = nil
+        Setting[:content_default_http_proxy] = "the proxy that shouldn't exist"
+        run_proxy_seed
+
+        assert_empty Setting[:content_default_http_proxy]
+      end
+
+      test "Creates proxy and assigns if not existing" do
+        refute ::HttpProxy.find_by(:name => 'foo.com')
+
+        SETTINGS[:katello][:cdn_proxy] = {
+          host: 'http://foo.com/',
```

Comment:
This might answer my question from above, but let me know if this is the expected format for host in theforeman/katello config.